session.rst

```diff
@@ -130,105 +391,303 @@ session metadata files:
+                    <call method="auth">
+                        <argument>%env(REDIS_USER)%</argument>
+                        <argument>%env(REDIS_PASSWORD)%</argument>
+                    </call> -->
                 </service>
+            </services>
+        </container>
 
-                <service id="session.namespacedattributebag"
-                    class="Symfony\Component\HttpFoundation\Session\Attribute\NamespacedAttributeBag"
-                />
+    .. code-block:: php
+
+        // ...
+        $container
+            // you can also use \RedisArray, \RedisCluster or \Predis\Client classes
+            ->register('Redis', \Redis::class)
+            ->addMethodCall('connect', ['%env(REDIS_HOST)%', '%env(int:REDIS_PORT)%'])
+            // uncomment the following if your Redis server requires a password:
+            // ->addMethodCall('auth', ['%env(REDIS_PASSWORD)%'])
+            // uncomment the following if your Redis server requires a user and a password (when user is not default):
+            // ->addMethodCall('auth', ['%env(REDIS_USER)%', '%env(REDIS_PASSWORD)%'])
+        ;
+
+Now pass this ``\Redis`` connection as an argument of the service associated to the
+:class:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\RedisSessionHandler`.
+This argument can also be a ``\RedisArray``, ``\RedisCluster``, ``\Predis\Client``,
+and ``RedisProxy``:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+            Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler:
+                arguments:
+                    - '@Redis'
+                    # you can optionally pass an array of options. The only options are 'prefix' and 'ttl',
+                    # which define the prefix to use for the keys to avoid collision on the Redis server
+                    # and the expiration time for any given entry (in seconds), defaults are 'sf_s' and null:
+                    # - { 'prefix': 'my_prefix', 'ttl': 600 }
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <services>
+            <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler">
+                <argument type="service" id="Redis"/>
+                <!-- you can optionally pass an array of options. The only options are 'prefix' and 'ttl',
+                     which define the prefix to use for the keys to avoid collision on the Redis server
+                     and the expiration time for any given entry (in seconds), defaults are 'sf_s' and null:
+                <argument type="collection">
+                    <argument key="prefix">my_prefix</argument>
+                    <argument key="ttl">600</argument>
+                </argument> -->
+            </service>
+        </services>
```

Comment:
```suggestion
        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd">
                
            <services>
                <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler">
                    <argument type="service" id="Redis"/>
                    <!-- you can optionally pass an array of options. The only options are 'prefix' and 'ttl',
                         which define the prefix to use for the keys to avoid collision on the Redis server
                         and the expiration time for any given entry (in seconds), defaults are 'sf_s' and null:
                    <argument type="collection">
                        <argument key="prefix">my_prefix</argument>
                        <argument key="ttl">600</argument>
                    </argument> -->
                </service>
            </services>
        </container>
```